[develop] Install kernel-devel for Rocky Linux from vault

enrico-usai opened the pull request:
### Description of changes

kernel-devel should be aligned to the installed version.
If there is kernel 8.8 we should have kernel-devel 8.8.

Without this patch we were installing latest kernel-devel package.
We cannot use `--releasever` to install 8.8 version because it is not more available in the repo.
Previous releases are moved into a vault area once a new minor release version is available for at least a week.

On RHEL8 instead all versions of the packages will remain available.

With this patch we're downloading this specific package from vault and installing it
if it's not available in the repository for the specific release version.

### References

Related to https://github.com/aws/aws-parallelcluster-cookbook/pull/2598

Commit:
Add an explicit step for updated headers installation

Signed-off-by: Enrico Usai <usai@amazon.com>

Commit:
Install kernel-devel for Rocky Linux from vault

kernel-devel should be aligned to the installed version.
If there is kernel 8.8 we should have kernel-devel 8.8.

Without this patch we were installing latest kernel-devel package.
We cannot use `--releasever` to install 8.8 version because it is not more available in the repo.
Previous releases are moved into a vault area once a new minor release version is available for at least a week.

On RHEL8 instead all versions of the packages will remain available.

With this patch we're downloading this specific package from vault and installing it
if it's not available in the repository for the specific release version.

Signed-off-by: Enrico Usai <usai@amazon.com>

jdeamicis commented on cli/src/pcluster/resources/imagebuilder/update_and_reboot.yaml:
```diff
@@ -256,15 +255,29 @@ phases:
+                # The same is done for Debian through `apt-get -y install linux-aws`
+                yum -y install kernel-devel
+              
+              elif [[ ${!PLATFORM} == DEBIAN ]]; then
```
If this is `InstallUpdatedHeaders`, I would expect only to be installing `linux-headers-aws` here. If you are adding this new step in the component, then let's add another one only for Debian derivatives where we install `linux-aws` and `linux-image-aws` and call it something like `InstallAWSDebianKernel` or something similar.

jdeamicis replied:
Ok, I consider this addressed.

jdeamicis reviewed (commented):
I believe we should add a separate step for the installation of the kernel packages (other than headers) for debian-based distributions.

Commit:
rocky-repos must be aligned with rocky-release

Error was:
```
yum update -y
Error:
 Problem: package rocky-repos-8.9-1.6.el8.noarch requires system-release = 8.9-1.6.el8, but none of the providers can be installed
  - cannot install the best update candidate for package rocky-repos-8.8-1.7.el8.noarch
  - package rocky-release-8.9-1.6.el8.noarch is filtered out by exclude filtering
```

Signed-off-by: Enrico Usai <usai@amazon.com>
